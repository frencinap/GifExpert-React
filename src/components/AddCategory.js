import React, { useState } from "react";
import PropTypes from 'prop-types'

//recibimos setCategories
export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState( '' )

    const handleInputValue = (e) => {
        setInputValue( e.target.value )
    }

    const handleSubmit = (e) => {
        //prevenir el comportamiento por defecto del submit
        e.preventDefault()
        //console.log('Submit Hecho');
        //validación submit
        if( inputValue.trim().length > 2) {
            setCategories( cats => [...cats, inputValue] )
            setInputValue('')
        }     
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputValue }
                placeholder="Agrega un héroe"
            />
        </ form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}