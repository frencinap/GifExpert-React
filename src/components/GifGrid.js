import React from 'react'

export const GifGrid = ({ category }) => {

    const getGif = async() =>{
        
        const url = 'https://api.giphy.com/v1/gifs/search?q=Avengers&limit=10&api_key=kQ50E0uTSpq7EGd7Nau8q6hA3EgxiN1P'
        const resp = await fetch( url )
        const { data } = await resp.json()

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    getGif()
    return (
        <div>
            <h3> { category } </h3>
        </div>
    )
}
