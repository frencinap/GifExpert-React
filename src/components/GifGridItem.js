import React from 'react'

//desestructuradas gracias al spreaoperator de GifGrid
export const GifGridItem = ( {id, title, url} ) => {

    console.log( id, title, url );
    return (
        <div className='card animate__animated animate__backInUp'>
            <img src={ url } alt={title} />
            <p>{ title } </p>
        </div>
    )
}
