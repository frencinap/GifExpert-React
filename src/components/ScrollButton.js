import React, { useState } from "react";
import { FiArrowUpCircle } from "react-icons/fi";
import { IconContext } from "react-icons";

export const ScrollButton = () => {

    const [visible, setVisible] = useState(false)

    const toggleVisible = () => {
        const scrolled = document.documentElement.scrollTop
        if( scrolled > 150 ){
            setVisible(true)
        }
        else if ( scrolled <= 150 ){
            setVisible(false)
        }

    }

    const scrollToTop = () => {
        window.scrollTo({
            top: 0,
            behavior: 'smooth',
        })
    }

    window.addEventListener('scroll', toggleVisible)

    return (
        <>
            <IconContext.Provider value={{ className: "arrow"}}>
                <FiArrowUpCircle
                style={{display: visible ? 'inline' : 'none'}}
                onClick={ scrollToTop } />
            </IconContext.Provider>
        </>    
    )
};
