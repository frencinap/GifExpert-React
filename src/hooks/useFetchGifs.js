//los hooks pueden tener su propio estado
import { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs'

export const useFetchGifs = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true,
    })

    //los efectos no son async pero puedo usar .then dentro de su argumento
    useEffect(()=> {
        getGifs( category )
            .then( imgs =>  {

                    setState({
                        data: imgs,
                        loading: false
                    })
          
            })
    }, [ category ])



    return state
}