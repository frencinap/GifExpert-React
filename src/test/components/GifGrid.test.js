import { shallow } from "enzyme";
import React from "react";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs} from '../../hooks/useFetchGifs'
// mock del hook
jest.mock('../../hooks/useFetchGifs')
 

describe('Pruebas en GifGrid component', () => {

    const category = 'Una categoria de ejemplo'
    //const wrapper = shallow( <GifGrid category={category}/> )

    test('Hacer match con el snapshot', () => {

        useFetchGifs.mockReturnValue({
            //estado incial del hook
            data: [],
            loading: true
        })
        const wrapper = shallow( <GifGrid category={category}/> )
        expect( wrapper ).toMatchSnapshot()
        
    });

    test('Debe de mostrar los items cuando se cargan imagenes con useFetch', () => {

        //mock del useFetchGifs, solo importamos el useFetchGifs
        const gifs = [{
            id: 'abcdef',
            url: 'www.panini.com/chupeteSuazo',
            title:'El Hombre del planeta gol'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        })
        const wrapper = shallow( <GifGrid category={category}/> )
        //expect( wrapper ).toMatchSnapshot()
        // al haber info el parrafo loading no deber√≠a mostrarse
        expect( wrapper.find('p').exists() ).toBe(false)
        //analizar que se muestre GifGridItem y el length de este sea igual a de gifs
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
    
    });
    
});