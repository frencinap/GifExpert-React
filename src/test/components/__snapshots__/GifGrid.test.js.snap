// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid component Hacer match con el snapshot 1`] = `
<Fragment>
  <h3>
     
    Una categoria de ejemplo
     
  </h3>
  <p
    className="animate__animated animate__flash animate__repeat-2"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
